src/packages/engine-core/src/NAPIEngine.ts

```diff
@@ -300,9 +301,8 @@ You may have to run ${chalk.greenBright(
     }
   }
 
-  private async runBeforeStop() {
-    await this.executingQueryPromise
```

Comment:
Hm, isn't this scary not to wait for an executing query to stop before we do this?

Reply:
Their is no need

Reply:
Why? Is it because it was only needed with binary because it is running "side-car"/separate process?
And Node-API is the same process? Though it should still be asynchronous and maybe needed no?

Reply:
Mainly because the if logic in the [beforeExitListener](https://github.com/prisma/prisma/pull/7740/files#diff-0e617d7e0e76eca325b6e16cf30d37d499ad42b8f61884a674260dd7d992c2a2R308) includes a query it will execute the [request](https://github.com/prisma/prisma/pull/7740/files#diff-0e617d7e0e76eca325b6e16cf30d37d499ad42b8f61884a674260dd7d992c2a2R402) which includes the required logic.

Reply:
i.e We don't need to wait for the `executingQueryPromise` here

Reply:
But `on('beforeExit')` is not the default / usually not used by users.

So by default `runBeforeExit()` does nothing since there will be no listener.

So the default behavior is different from NodeEngine?

In NodeEngine I see "$disconnect" call "stop" which await the currentRequestPromise before stopping
https://github.com/prisma/prisma/blob/feat/napi/beforeExit/src/packages/engine-core/src/NodeEngine.ts#L816-L835

Reply:
> So by default runBeforeExit() does nothing since there will be no listener.
Correct

> So the default behavior is different from NodeEngine?
No, it is the same you can see that [here](https://github.com/prisma/prisma/blob/master/src/packages/engine-core/src/NAPIEngine.ts#L343)  stop awaits the `executingQueryPromise`

Reply:
Ok Thanks for the details!